src/formatDate.js

```diff
@@ -50,7 +50,45 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [ a, b, c, separator ] = fromFormat;
+  const [ , , , newSeparator ] = toFormat;
+
+  const newData = [];
+
+  const arr = date.split(separator);
+
+  const obj = {};
+
+  obj[a] = arr[0];
+  obj[b] = arr[1];
+  obj[c] = arr[2];
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    obj.YYYY = (obj.YYYY).substring(2);
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (obj.YY < 30) {
+      obj.YYYY = `20${obj.YY}`;
+    } else {
+      obj.YYYY = `19${obj.YY}`;
+    }
+  }
```

Comment:
Try to combine these two conditions, cause they are the same.

Reply:
Could you please show me how to combine this cases. I optimized them with the ternary operator, but still two checks remain: 
if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
    obj.YYYY = (obj.YYYY).substring(2);
  }

  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
    obj.YYYY = obj.YY < 30 ? `20${obj.YY}` : `19${obj.YY}`;
  }

Reply:
> Try to combine these two conditions, cause they are the same.
Could you please show me how to combine this cases. I optimized them with the ternary operator, but still two checks remain:
if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
obj.YYYY = (obj.YYYY).substring(2);
}

if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
obj.YYYY = obj.YY < 30 ? 20${obj.YY} : 19${obj.YY};
}

Reply:
> Try to combine these two conditions, cause they are the same.

Could you please show me how to combine this cases. I optimized them with the ternary operator, but still two checks remain:
if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
obj.YYYY = (obj.YYYY).substring(2);
}

if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
obj.YYYY = obj.YY < 30 ? 20${obj.YY} : 19${obj.YY};
}